src/d3a/d3a_core/sim_results/export_unmatched_loads.py

```diff
@@ -15,127 +15,160 @@
-            # Recurse even further. Merge new results with existing ones
-            unmatched_loads = {**unmatched_loads,
-                               **_recurse_area_tree(child, area_select_function)}
-    return unmatched_loads
-
-
-def select_house_or_cell_tower(area):
-    return _is_house_node(area) or _is_cell_tower_node(area)
 
-
-def select_house_or_load(area):
-    return _is_house_node(area) or isinstance(area.strategy, LoadHoursStrategy)
+def hour_list():
+    if GlobalConfig.sim_duration > duration(days=1):
+        return [GlobalConfig.start_date.add(days=day, hours=hour)
+                for day, hour in product(range(GlobalConfig.sim_duration.days), range(24))]
+    else:
+        return [GlobalConfig.start_date.add(hours=hour) for hour in range(24)]
 
 
-def export_unmatched_loads(area, all_devices=False):
-    unmatched_loads_result = {}
+class ExportUnmatchedLoads:
+    def __init__(self, area):
 
-    area_select_function = select_house_or_load if all_devices else select_house_or_cell_tower
-    area_tree = _recurse_area_tree(area, area_select_function)
-    # Calculate overall metrics for the whole grid
-    unmatched_loads_result["unmatched_load_count"] = \
-        sum([v["unmatched_load_count"] for k, v in area_tree.items()])
-    unmatched_loads_result["all_loads_met"] = (unmatched_loads_result["unmatched_load_count"] == 0)
-    area_tree = OrderedDict(sorted(area_tree.items()))
-    unmatched_loads_result["areas"] = area_tree
-    return unmatched_loads_result
+        self.hour_list = hour_list()
+        # This is for only returning data until the current time_slot:
+        if hasattr(area, "past_markets") and len(list(area.past_markets)) > 0:
+            self.latest_time_slot = list(area.past_markets)[-1].time_slot
+        else:
+            self.latest_time_slot = self.hour_list[0]
+        self.name_uuid_map = {area.name: area.uuid}
+        self.area = area
+
+    def __call__(self):
+
+        unmatched_loads = self.arrange_output(
+            self.expand_to_ul_to_hours(
+                self.expand_ul_to_parents(
+                    self.find_unmatched_loads(self.area, {})[self.area.name],
+                    self.area.name, {})),
+            self.area)
+
+        return unmatched_loads, self.change_name_to_uuid(unmatched_loads)
+
+    def find_unmatched_loads(self, area, indict):
+        """
+        Aggregates list of times for each unmatched time slot for each load
+        """
+        indict[area.name] = {}
+        for child in area.children:
+            self.name_uuid_map[child.name] = child.uuid
+            if child.children:
+                indict[area.name] = self.find_unmatched_loads(child, indict[area.name])
+            else:
+                if isinstance(child.strategy, LoadHoursStrategy):
+                    indict[area.name][child.name] = self._accumulate_unmatched_loads(child)
+        return indict
+
+    @classmethod
+    def _accumulate_unmatched_loads(cls, area):
```

Comment:
Trivial, but this method does not actually accumulate, but just calculates the unmatched load for a leaf device. I would prefer if it was named similar to '_calculate_unmatched_loads_leaf_area' or similar